### PR TITLE
bitcoin-core: Unlock all fuzz targets

### PR DESCRIPTION
**Pull request by maflcko**
Second attempt:
* CI is expected to run out of disk space for now
* The python one-liner is ugly

### CODE REVIEW

File: projects/bitcoin-core/build.sh [hunk 1]
```diff
@@ -47,13 +49,18 @@ CONFIG_SITE="$PWD/depends/$BUILD_TRIPLET/share/config.site" ./configure --enable
 
 make -j$(nproc)
 
-# Limit to a few targets as temporary workaround for https://github.com/google/oss-fuzz/pull/5699#issuecomment-831030305
-FUZZ_TARGETS=( 'process_messages' 'asmap' )
+WRITE_ALL_FUZZ_TARGETS_AND_ABORT="/tmp/a" "./src/test/fuzz/fuzz" || true
```
Comment (jonathanmetzman) [inline projects/bitcoin-core/build.sh:52]: Is this just for testing?
What's the purpose of these `|| true` here and on 42?

File: projects/bitcoin-core/build.sh [hunk 2]
```diff
@@ -47,13 +49,18 @@ CONFIG_SITE="$PWD/depends/$BUILD_TRIPLET/share/config.site" ./configure --enable
...
+#FUZZ_TARGETS=( 'process_messages' 'asmap' )
+export MAGIC_STR="b5813eee2abc9d3358151f298b75a72264ffa119d2f71ae7fefa15c4b70b4bc5b38e87e3107a730f25891ea428b2b4fabe7a84f5bfa73c79e0479e085e4ff157"
+sed -i "s|std::getenv(\"FUZZ\")|\"$MAGIC_STR\"|g" "./src/test/fuzz/fuzz.cpp"
+make -j$(nproc)
 for fuzz_target in ${FUZZ_TARGETS[@]}; do
-  git checkout -- "./src/test/fuzz/fuzz.cpp"
-  sed -i "s|std::getenv(\"FUZZ\")|\"$fuzz_target\"|g" "./src/test/fuzz/fuzz.cpp"
-  make -j$(nproc)
-  mv ./src/test/fuzz/fuzz $OUT/$fuzz_target
+  python3 -c "c_str_target=b\"${fuzz_target}\x00\";c_str_magic=b\"$MAGIC_STR\";c=open('./src/test/fuzz/fuzz','rb').read();c=c.replace(c_str_magic, c_str_target+c_str_magic[len(c_str_target):]);open(\"$OUT/$fuzz_target\",'wb').write(c)"
```
Comment (jonathanmetzman) [inline projects/bitcoin-core/build.sh:62]: :-)

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Comment by jonathanmetzman** [comment]
Ouch so it runs out of space anyway?

**Comment by MarcoFalke** [comment]
Let me try something to reduce the disk space ...

File: projects/bitcoin-core/build.sh (hunk 1 above)
Comment (MarcoFalke) [reply projects/bitcoin-core/build.sh:50]: Removed the one in line 42 by merging the upstream patch

**Comment by MarcoFalke** [comment]
Not sure why afl doesn't have any log output, but I think you can merge this

**Pull request merged**